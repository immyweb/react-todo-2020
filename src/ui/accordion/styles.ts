import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
