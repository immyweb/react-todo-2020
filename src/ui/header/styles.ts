import styled from '@emotion/styled';

export const HeaderContainer = styled.header`
  margin-bottom: 25px;
`;

export const Heading = styled.h1`
  margin: 0;
`;
