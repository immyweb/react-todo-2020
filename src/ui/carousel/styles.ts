import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
