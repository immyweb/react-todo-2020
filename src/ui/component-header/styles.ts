import glamorous from 'glamorous';

import { P, H2 } from '../typography/typography';
import { colours } from '../../styles/colours';

import { mediaQueries } from '../../styles/breakpoints';
import { IAlign } from '../../utils/alignment-utils';

interface IGProps {
  inline?: boolean;
  alignment?: IAlign;
  linkColour?: string;
  background?: boolean;
  hasBackground?: boolean;
}

const styles = {
  fontSizeLarge: 28,
  fontSizeSmall: 24,
  inlineHighlight: 16,
  marginBottomLarge: 12,
  h2LineHeight: 1.14,
  pLineHeight: 1.38,
};

export const TitleWrapper = glamorous.div<IGProps>(
  {
    label: 'title-wrapper',
  },
  ({ alignment, background, theme }: any) => ({
    textAlign: alignment.x,
    background: background ? theme.primary : 'none',
    padding: background ? '13px 20px 13px 15px' : 0,
    borderLeftStyle: background ? 'solid' : 'none',
    borderLeftWidth: '5px',
    borderLeftColor: theme.secondary,
  }),
);

export const Title = glamorous(H2)<IGProps>(
  {
    label: 'title',
    fontSize: styles.fontSizeSmall,
    lineHeight: styles.h2LineHeight,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      lineHeight: styles.h2LineHeight,
    },
  },
  ({ inline, alignment, background }: any) => ({
    display: inline ? 'inline-block' : 'block',
    textTransform: background ? 'unset' : inline ? 'unset' : 'uppercase',
    marginBottom: background ? 0 : styles.marginBottomLarge,
    textAlign: alignment.x,
    color: background ? colours.$white : colours.$offBlack,
  }),
);

export const Highlight = glamorous(Title)<IGProps>(
  {
    label: 'highlight',
    marginBottom: 0,
    fontSize: styles.fontSizeSmall,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      marginBottom: 0,
    },
  },
  ({ inline, alignment, fontColour, background }: any) => ({
    fontSize: inline ? styles.fontSizeSmall : styles.inlineHighlight,
    textTransform: 'uppercase',
    textAlign: alignment.x,
    color: fontColour,
    marginRight: inline ? 4 : 0,
    [mediaQueries.tabletPortrait]: {
      fontSize: inline ? styles.fontSizeLarge : styles.inlineHighlight,
    },
  }),
);

export const SubTitle = glamorous(P)<IGProps>(
  {
    label: 'sub-title',
    lineHeight: styles.pLineHeight,
    textTransform: 'lowercase',
    marginBottom: 20,
    [mediaQueries.tabletPortrait]: {
      lineHeight: styles.pLineHeight,
    },
  },
  ({ alignment, linkColour, hasBackground }: any) => ({
    textAlign: alignment.x,
    marginTop: hasBackground ? 20 : 0,
    '& a': {
      color: linkColour,
      textDecoration: 'none',
    },

    '& a:hover': {
      textDecoration: 'underline',
    },
  }),
);
