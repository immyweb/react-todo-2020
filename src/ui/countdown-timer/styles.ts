import glamorous from 'glamorous';
import { H1, H2, P } from '../typography/typography';
import { mediaQueries } from '../../styles/breakpoints';
import { fonts } from '../../styles/fonts';

import { IAlign } from '../../utils/alignment-utils';

export const CountdownWrapper = glamorous.div<{
  backgroundImage?: string;
}>(
  {
    position: 'relative',
    width: '100%',
    height: '100%',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    backgroundPosition: 'center center',
    minHeight: 185,
  },
  ({ backgroundImage }) => ({
    backgroundImage: backgroundImage ? `url(${backgroundImage})` : '',
  }),
);

export const CountdownContent = glamorous.div<{ alignment?: IAlign }>(
  {
    position: 'absolute',
    width: '100%',
    padding: 10,
  },
  ({ alignment }) => ({
    top: alignment ? alignment.y : 0,
    left: alignment ? alignment.x : 0,
    transform: `translate(-${alignment ? alignment.x : 0}, -${alignment ? alignment.y : 0})`,
  }),
);

export const CountdownFrame = glamorous.div({
  minHeight: 80,
  position: 'relative',
});

export const CountdownAligner = glamorous.div<{ alignment?: IAlign }>(
  {
    position: 'absolute',
    width: '100%',
    top: '50%',
  },
  ({ alignment }) => ({
    left: alignment ? alignment.x : 0,
    transform: `translate(-${alignment ? alignment.x : 0}, -50%)`,
  }),
);

export const HeaderWrapper = glamorous.div({
  label: 'header-wrapper',
});

export const Title = glamorous(H2)<{ alignment?: IAlign; textColour: string }>(
  {
    marginBottom: 2,
    textTransform: 'uppercase',
    [mediaQueries.tabletPortrait]: {
      lineHeight: 1,
    },
  },
  ({ alignment, textColour }: any) => ({
    textAlign: alignment ? alignment.x : 'left',
    color: textColour,
  }),
);

export const SubTitle = glamorous(P)<{
  alignment?: IAlign;
  textColour: string;
}>(
  {
    fontSize: 14,
    lineHeight: 1.25,
    marginBottom: 5,
    [mediaQueries.tabletPortrait]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
  },
  ({ alignment, textColour }: any) => ({
    textAlign: alignment ? alignment.x : 'left',
    color: textColour,
  }),
);

export const UnitWrapper = glamorous.div<{
  alignment?: IAlign;
  opacity?: number;
}>(
  {
    transition: `opacity ${500}ms linear`,
    transitionDelay: `${1000}ms`,
    ':nth-child(n) > div': {
      marginRight: 3,
    },
  },
  ({ alignment, opacity }: any) => ({
    opacity: opacity ? opacity : 0,
    textAlign: alignment ? alignment.x : 'left',
  }),
);

export const UnitInstance = glamorous.div({
  display: 'inline-block',
});

export const UnitValue = glamorous(H1)<{ textColour: string }>(
  {
    fontSize: 42,
    lineHeight: 1.17,
    fontFamily: fonts.$theSunHeavyNarrow,
    textAlign: 'center',
    marginBottom: 0,
    minWidth: 47,
    [mediaQueries.tabletPortrait]: {
      fontSize: 48,
      lineHeight: 1.17,
    },
    [mediaQueries.mobile]: {
      minWidth: 62,
      fontSize: 48,
      lineHeight: 1.17,
    },
  },
  ({ textColour }) => ({
    color: textColour,
  }),
);

export const UnitLabel = glamorous(P)<{ textColour: string }>(
  {
    fontSize: 12,
    lineHeight: 1.25,
    marginBottom: 0,
    textAlign: 'center',
    [mediaQueries.tabletPortrait]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
    [mediaQueries.mobile]: {
      fontSize: 14,
      lineHeight: 1.25,
    },
  },
  ({ textColour }) => ({
    color: textColour,
  }),
);

export const DigitContainer = glamorous.div({
  width: 29,
  display: 'inline-block',
  textAlign: 'center',
});
