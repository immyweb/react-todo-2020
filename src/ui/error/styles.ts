import glamorous from 'glamorous';

import { H3 } from '../typography/typography';
import { fonts } from '../../styles/fonts';

interface IGProps {
  backgroundColour?: string;
  marginBottom?: number;
}

export const ErrorWrapper = glamorous.div<IGProps>(
  {
    label: 'error-wrapper',
    position: 'relative',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 200,
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const Heading = glamorous(H3)<IGProps>(
  {
    fontFamily: fonts.$theSunHeavyCondensed,
    textAlign: 'center',
    fontSize: 22,
  },
  ({ marginBottom }) => ({
    marginBottom: marginBottom ? marginBottom : 5,
  }),
);

export const Graphic = glamorous.div({
  label: 'graphic',
  position: 'relative',
  margin: '0 auto 10px auto',
});
