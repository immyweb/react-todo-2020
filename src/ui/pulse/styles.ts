import glamorous from 'glamorous';

import { ArrowPosition } from '../label/label';
import { keyframes } from 'glamor';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';

const pulsate = () => {
  return keyframes({
    '0%': { opacity: 0, transform: 'scale(0.1, 0.1)' },
    '50%': { opacity: 1 },
    '100%': { opacity: 0, transform: 'scale(1.2, 1.2)' },
  });
};

interface IGProps {
  labelSide?: string;
  labelPosition?: string;
  colour?: string;
  withMap?: boolean;
  size?: number;
}

export const PulseContainer = glamorous.div<IGProps>(
  {
    label: 'pulse-container',
    position: 'absolute',
    top: 0,
    width: '100%',
    height: '100%',
    zIndex: 1,
  },
  ({ withMap }) => ({
    transform: withMap ? `rotateX(55deg)` : 'rotateX(0deg)',
  }),
);

export const PulseInnerContainer = glamorous.div({
  label: 'pulse-inner-container',
  width: 38,
  height: 38,
  [mediaQueries.tabletPortrait]: {
    width: 48,
    height: 48,
  },
});

export const PulseInner = glamorous.div<IGProps>(
  {
    label: 'pulse-inner',
    borderRadius: '50%',
    position: 'absolute',
    animation: `${pulsate()} 1s ease-out`,
    animationIterationCount: 'infinite',
    animationDelay: '1.1s',
    opacity: 0,
    width: 30,
    height: 30,
    [mediaQueries.tabletPortrait]: {
      width: 40,
      height: 40,
    },
  },
  ({ colour }) => ({
    boxShadow: colour ? `0 0 1px 2px ${colour}` : `0 0 1px 2px ${colours.$main}`,
  }),
);

export const PulseNoLabel = glamorous(PulseInner)(
  {
    label: 'pulse-no-label',
    position: 'relative',
  },
  ({ size }) => ({
    height: size ? size : 30,
    width: size ? size : 30,
    [mediaQueries.tabletPortrait]: {
      height: size ? size : 40,
      width: size ? size : 40,
    },
  }),
);

export const PulseTop = glamorous(PulseInner)(
  {
    label: 'pulse-top',
  },
  ({ labelPosition, withMap }) => ({
    top: withMap ? -50 : -22,
    transform: labelPosition === ArrowPosition.CENTER ? 'translateX(-50%)' : 'translateX(0%)',
    left:
      labelPosition === ArrowPosition.LEFT
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? 'calc(50% - 15px)'
        : 'auto',
    right:
      labelPosition === ArrowPosition.RIGHT
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? '50%'
        : 'auto',
    [mediaQueries.tabletPortrait]: {
      top: withMap ? -50 : -32,
      left:
        labelPosition === ArrowPosition.LEFT
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? 'calc(50% - 20px)'
          : 'auto',
      right:
        labelPosition === ArrowPosition.RIGHT
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? '50%'
          : 'auto',
    },
  }),
);

export const PulseBottom = glamorous(PulseInner)(
  {
    label: 'pulse-bottom',
  },
  ({ labelPosition, withMap }) => ({
    bottom: withMap ? -50 : -22,
    transform: labelPosition === ArrowPosition.CENTER ? 'translateX(-50%)' : 'translateX(0%)',
    left:
      labelPosition === ArrowPosition.LEFT
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? 'calc(50% - 15px)'
        : 'auto',
    right:
      labelPosition === ArrowPosition.RIGHT
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? '50%'
        : 'auto',
    [mediaQueries.tabletPortrait]: {
      bottom: withMap ? -50 : -32,
      left:
        labelPosition === ArrowPosition.LEFT
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? 'calc(50% - 20px)'
          : 'auto',
      right:
        labelPosition === ArrowPosition.RIGHT
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? '50%'
          : 'auto',
    },
  }),
);

export const PulseLeft = glamorous(PulseInner)(
  {
    label: 'pulse-left',
  },
  ({ labelPosition, withMap }) => ({
    left: withMap ? -35 : -22,
    transform: labelPosition === ArrowPosition.CENTER ? 'translateY(-50%)' : 'translateY(0%)',
    top:
      labelPosition === ArrowPosition.TOP
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? 'calc(50% - 15px)'
        : 'auto',
    bottom:
      labelPosition === ArrowPosition.BOTTOM
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? '50%'
        : 'auto',
    [mediaQueries.tabletPortrait]: {
      left: withMap ? -35 : -32,
      top:
        labelPosition === ArrowPosition.TOP
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? 'calc(50% - 20px)'
          : 'auto',
      bottom:
        labelPosition === ArrowPosition.BOTTOM
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? '50%'
          : 'auto',
    },
  }),
);

export const PulseRight = glamorous(PulseInner)(
  {
    label: 'pulse-right',
  },
  ({ labelPosition, withMap }) => ({
    right: withMap ? -35 : -22,
    transform: labelPosition === ArrowPosition.CENTER ? 'translateY(-50%)' : 'translateY(0%)',
    top:
      labelPosition === ArrowPosition.TOP
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? 'calc(50% - 15px)'
        : 'auto',
    bottom:
      labelPosition === ArrowPosition.BOTTOM
        ? 5
        : labelPosition === ArrowPosition.CENTER
        ? '50%'
        : 'auto',
    [mediaQueries.tabletPortrait]: {
      right: withMap ? -35 : -32,
      top:
        labelPosition === ArrowPosition.TOP
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? 'calc(50% - 20px)'
          : 'auto',
      bottom:
        labelPosition === ArrowPosition.BOTTOM
          ? 2
          : labelPosition === ArrowPosition.CENTER
          ? '50%'
          : 'auto',
    },
  }),
);
