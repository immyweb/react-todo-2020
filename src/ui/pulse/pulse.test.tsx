import * as React from 'react';
import { shallow } from 'enzyme';

import Pulse from './pulse';
import { ArrowPosition } from '../label/label';
import PulseLabel from './pulse-label';
import { colours } from '../../styles/colours';

function shallowPulse(props?: any) {
  return shallow(<Pulse {...props}>Lovely label</Pulse>);
}

function shallowPulseLabel(props?: any) {
  return shallow(
    <PulseLabel labelSide={ArrowPosition.TOP} labelPosition={ArrowPosition.RIGHT} {...props}>
      Lovely label
    </PulseLabel>,
  );
}

describe('<Pulse />', () => {
  it('renders correctly with default values', () => {
    const wrapper = shallowPulse();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with colour override', () => {
    const wrapper = shallowPulse({ colour: colours.$fabulous });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with size override', () => {
    const wrapper = shallowPulse({ size: 150 });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<PulseLabel />', () => {
  it('renders correctly with default values', () => {
    const wrapper = shallowPulseLabel();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with Map', () => {
    const wrapper = shallowPulseLabel({ withMap: true });
    expect(wrapper).toMatchSnapshot();
  });
});
