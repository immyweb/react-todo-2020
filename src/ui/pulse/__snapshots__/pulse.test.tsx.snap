// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pulse /> renders correctly with colour override 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pulse
    colour="#e665bf"
  >
    Lovely label
  </Pulse>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bounds": true,
      "children": [Function],
      "onResize": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bounds": true,
        "children": [Function],
        "onResize": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Pulse /> renders correctly with default values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pulse>
    Lovely label
  </Pulse>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bounds": true,
      "children": [Function],
      "onResize": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bounds": true,
        "children": [Function],
        "onResize": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Pulse /> renders correctly with size override 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pulse
    size={150}
  >
    Lovely label
  </Pulse>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bounds": true,
      "children": [Function],
      "onResize": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bounds": true,
        "children": [Function],
        "onResize": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PulseLabel /> renders correctly with Map 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PulseLabel
    labelPosition="RIGHT"
    labelSide="TOP"
    withMap={true}
  >
    Lovely label
  </PulseLabel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <glamorous(glamorous(div))
          labelPosition="RIGHT"
          withMap={true}
        />
      </div>,
      "className": "pulse",
      "withMap": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <glamorous(glamorous(div))
            labelPosition="RIGHT"
            withMap={true}
          />,
          false,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "colour": undefined,
            "labelPosition": "RIGHT",
            "withMap": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        false,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <glamorous(glamorous(div))
            labelPosition="RIGHT"
            withMap={true}
          />
        </div>,
        "className": "pulse",
        "withMap": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(div))
              labelPosition="RIGHT"
              withMap={true}
            />,
            false,
            false,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "colour": undefined,
              "labelPosition": "RIGHT",
              "withMap": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<PulseLabel /> renders correctly with default values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PulseLabel
    labelPosition="RIGHT"
    labelSide="TOP"
  >
    Lovely label
  </PulseLabel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <glamorous(glamorous(div))
          labelPosition="RIGHT"
          withMap={false}
        />
      </div>,
      "className": "pulse",
      "withMap": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <glamorous(glamorous(div))
            labelPosition="RIGHT"
            withMap={false}
          />,
          false,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "colour": undefined,
            "labelPosition": "RIGHT",
            "withMap": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        false,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <glamorous(glamorous(div))
            labelPosition="RIGHT"
            withMap={false}
          />
        </div>,
        "className": "pulse",
        "withMap": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(div))
              labelPosition="RIGHT"
              withMap={false}
            />,
            false,
            false,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "colour": undefined,
              "labelPosition": "RIGHT",
              "withMap": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
